### ISSUE TITLE
Review repetive method calls with many parameters

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/gammasim-tools#309]
"Rich" (many parameters) and repetitive method calls makes the code sometimes hard to read (as it is e.g., hard to see if there is something different in the list of arguments or not). Wondering if there isn't a better solution, like working with dictionaries or so. 

Try and understand if there is another solution.

Example:
```
132     def getLogFile(self, run):
133         """Get full path of the simtel log file for a given run."""
134         fileName = names.simtelLogFileName(
135             run=run,
136             primary=self.config.primary,
137             arrayName=self.arrayModel.layoutName,
138             site=self.arrayModel.site,
139             zenith=self.config.zenithAngle,
140             azimuth=self.config.azimuthAngle,
141             label=self.label,
142         )
143         return self._simtelLogDir.joinpath(fileName)
144
145     def getSubLogFile(self, run, mode='out'):
146         ''' Get full path of the simtel log file for a given run. '''
147         fileName = names.simtelSubLogFileName(
148             run=run,
149             primary=self.config.primary,
150             arrayName=self.arrayModel.layoutName,
151             site=self.arrayModel.site,
152             zenith=self.config.zenithAngle,
153             azimuth=self.config.azimuthAngle,
154             label=self.label,
155             mode=mode
156         )
157         return self._simtelLogDir.joinpath(fileName)
158
159     def getHistogramFile(self, run):
160         """Get full path of the simtel histogram file for a given run."""
161         fileName = names.simtelHistogramFileName(
162             run=run,
163             primary=self.config.primary,
164             arrayName=self.arrayModel.layoutName,
165             site=self.arrayModel.site,
166             zenith=self.config.zenithAngle,
167             azimuth=self.config.azimuthAngle,
168             label=self.label,
169         )
170         return self._simtelDataDir.joinpath(fileName)
```

**Comment by GernotMaier** [comment]
I think this has been mostly addressed with #332 and #330

**Comment by GernotMaier** [comment]
You are correct - the changes required are mostly in simtools/util/names.py (I've looked in the wrong file).

Let's do that as well for 0.4.
